File: ios/HaapiModule/HaapiModule.m
//
//  HaapiModule.m
//  HaapiModule
//
//  Created by Daniel Lindau on 2023-12-28.
//


#import <React/RCTBridgeModule.h>
#import <React/RCTEventEmitter.h>

@interface RCT_EXTERN_MODULE(NetworkStatus, RCTEventEmitter)

RCT_EXTERN_METHOD(start)

@end

File: ios/HaapiModule.swift
//
//  HaapiModule.swift
//  HaapiModule
//
//  Created by Daniel Lindau on 2023-12-28.
//

import Foundation
import Network

@objc(HaapiModule)
class HaapiModule: RCTEventEmitter {

  override init() {
    super.init()
  }

  override func supportedEvents() -> [String]! {
    return ["HaapiError"]
  }

  @objc
  func start() {
    self.sendEvent(withName: "HaapiError", body: ["error": "HaapiError", "error_description": "Not Implemented"])
  }
  

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
